lib/approval_request_listener.rb

```diff
@@ -31,9 +31,12 @@ def subscribe_to_approval_updates
 
   def process_event(topic)
     approval = ApprovalRequest.find_by!(:approval_request_ref => topic.payload["request_id"])
-    approval.order_item.update_message("info", "Task update message received with payload: #{topic.payload}")
+    Rails.logger.info("Task update message received with payload: #{topic.payload}")
+    approval.order_item.update_message("info", "Approval #{approval.id} #{topic.payload['decision']}")
+
     if topic.message == EVENT_REQUEST_FINISHED
       update_and_log_state(approval, topic.payload)
+      Catalog::OrderItemTransition.new(approval.order_item_id).process
```

Comment:
Should we rescue and log exception(s) from this service?

Reply:
Sure, I'll rescue the `StandardError` that `Catalog::SubmitOrder` can raise, and handle it appropriately.